game/factions/faction.py

```diff
@@ -0,0 +1,216 @@
+import logging
+from dataclasses import dataclass
+from typing import Optional
+
+import dcs
+from dcs.planes import PlaneType, plane_map
+from dcs.unittype import VehicleType, UnitType
+from dcs.vehicles import vehicle_map, Armor, Unarmed, Infantry, Fortification, Artillery, AirDefence
+
+from game.data.building_data import WW2_ALLIES_BUILDINGS, DEFAULT_AVAILABLE_BUILDINGS, WW2_GERMANY_BUILDINGS
+from game.data.doctrine import Doctrine, MODERN_DOCTRINE, COLDWAR_DOCTRINE, WWII_DOCTRINE
+from pydcs_extensions.mod_units import MODDED_VEHICLES, MODDED_AIRPLANES
+
+
+@dataclass
+class Faction:
+
+    # Country used by this faction
+    country: str
+
+    # Nice name of the faction
+    name: str
+
+    # Available aircraft
+    aircrafts: [UnitType]
+
+    # Available awacs aircraft
+    awacs: [UnitType]
+
+    # Available tanker aircraft
+    tankers: [UnitType]
+
+    # Available frontline units
+    frontline_units: [VehicleType]
+
+    # Available artillery units
+    artillery_units: [VehicleType]
+
+    # Infantry units used
+    infantry_units: [VehicleType]
+
+    # Logistics units used
+    logistics_units: [VehicleType]
+
+    # List of units that can be deployed as SHORAD
+    shorads: [str]
+
+    # Possible SAMS site generators for this faction
+    sams: [str]
+
+    # Possible Missile site generators for this faction
+    missiles: [str]
+
+    # Required mods or asset packs
+    requirements: {str: str}
+
+    # possible aircraft carrier units
+    aircraft_carrier: [UnitType]
+
+    # possible helicopter carrier units
+    helicopter_carrier: [UnitType]
+
+    # Possible carrier names
+    carrier_names: [str]
+
+    # Possible helicopter carrier names
+    helicopter_carrier_names: [str]
+
+    # Navy group generators
+    navy_generators: [str]
+
+    # Available destroyers
+    destroyers: [str]
+
+    # Available cruisers
+    cruisers: [str]
+
+    # How many navy group should we try to generate per CP on startup for this faction
+    navy_group_count: int
+
+    # How many missiles group should we try to generate per CP on startup for this faction
+    missiles_group_count: int
+
+    # Whether this faction has JTAC access
+    has_jtac: bool
+
+    # Unit to use as JTAC for this faction
+    jtac_unit: str
+
+    # doctrine
+    doctrine: Doctrine
+
+    # List of available buildings for this faction
+    building_set: [str]
+
+    def __init__(self):
+        self.country = ""
```

Comment:
If you make all of these parameters, you can delete the `__init__` and make the `dataclass` frozen. https://docs.python.org/3/library/dataclasses.html#dataclasses.field lets you set up defaults where needed.